ChronoKeeper/ChronoKeeperInstance.cpp

```diff
@@ -252,28 +273,26 @@ int main(int argc, char**argv)
         sleep(30);
     }
 
-    //unregister from the chronoVisor so that no new story requests would be coming
+    /// Unregister from ChronoVisor ____________________________________________________________________________________
+    // Unregister from the chronoVisor so that no new story requests would be coming
     keeperRegistryClient->send_unregister_msg(keeperIdCard);
     delete keeperRegistryClient;
 
-    //stop recording events
+    /// Stop services and shut down ____________________________________________________________________________________
+    Logger::getLogger()->info("[ChronoKeeperInstance] Initiating shutdown procedures.");
+    // Stop recording events
     delete keeperRecordingService;
     delete keeperDataAdminService;
-
-    //shutdown the Data Collection
+    // Shutdown the Data Collection
     theDataStore.shutdownDataCollection();
-
-    // shutdown extraction module
+    // Shutdown extraction module
     // drain extractionQueue and stop extraction xStreams
     storyExtractor.shutdownExtractionThreads();
-
     // these are not probably needed as thalium handles the engine finalization...
     //  recordingEngine.finalize();
     //  collectionEngine.finalize();
-
     delete recordingEngine;
     delete dataAdminEngine;
-
+    Logger::getLogger()->info("[ChronoKeeperInstance] Shutdown completed. Exiting.");
```

Comment:
I have chosen to keep the destructor private because the class is managed by a std::shared_ptr. This approach automatically handles the logger's lifecycle, including its destruction when the application terminates or when all references are out of scope. This design aligns with the singleton pattern, ensuring controlled and safe management of the logger's instance without requiring public access to the destructor. In any case, do you prefer to make it public?